File: clients/python/uiform/types/schemas/generate.py
from typing import Any, Literal
from openai.types.chat.chat_completion_reasoning_effort import ChatCompletionReasoningEffort
from pydantic import BaseModel, Field

from ..mime import MIMEData
from ..modalities import Modality


class GenerateSchemaRequest(BaseModel):
    documents: list[MIMEData]
    model: str = "gpt-4o-mini"
    temperature: float = 0.0
    reasoning_effort: ChatCompletionReasoningEffort = "medium"
    modality: Modality
    instructions: str | None = None
    """The modality of the document to load."""
    image_resolution_dpi: int = Field(default=96, description="Resolution of the image sent to the LLM")
    browser_canvas: Literal['A3', 'A4', 'A5'] = Field(default='A4', description="Sets the size of the browser canvas for rendering documents in browser-based processing. Choose a size that matches the document type.")

    """The image operations to apply to the document."""

    stream: bool = False
    """Whether to stream the response."""


class GenerateSystemPromptRequest(GenerateSchemaRequest):
    """
    The request body for generating a system prompt for a JSON Schema.
    """

    json_schema: dict[str, Any]
    

File: clients/node/src/types/schemas/generate.ts
import { z } from 'zod';
import { MIMEDataSchema } from '../mime.js';
import { ModalitySchema } from '../modalities.js';
import { BrowserCanvasSchema } from '../browser_canvas.js';
import { ChatCompletionReasoningEffortSchema } from '../documents/extractions.js';

export const GenerateSchemaRequestSchema = z.object({
  documents: z.array(MIMEDataSchema),
  model: z.string().default('gpt-4o-mini'),
  temperature: z.number().default(0.0),
  reasoning_effort: ChatCompletionReasoningEffortSchema.default('medium'),
  modality: ModalitySchema,
  instructions: z.string().nullable().optional(),
  image_resolution_dpi: z.number().int().default(96),
  browser_canvas: BrowserCanvasSchema.default('A4'),
  stream: z.boolean().default(false),
});
export type GenerateSchemaRequest = z.infer<typeof GenerateSchemaRequestSchema>;

export const GenerateSystemPromptRequestSchema = GenerateSchemaRequestSchema.extend({
  json_schema: z.record(z.any()),
});
export type GenerateSystemPromptRequest = z.infer<typeof GenerateSystemPromptRequestSchema>;